src/saml.py

```diff
@@ -29,9 +40,9 @@ def __init__(self, charm_state: CharmState):
         """
         self._charm_state = charm_state
 
-    @cached_property
-    def _tree(self) -> "etree.ElementTree":  # type: ignore
-        """Fetch metadata contents.
+    @property
+    def _read_tree(self) -> "etree.ElementTree":
```

Comment:
Perhaps this would be better as  a function now?

Reply:
Indeed